forem/forem
# First working version to integrate native bridge image upload

## What type of PR is this? (check all applicable)


## Description

This feature will ideally enhance the native mobile experience of uploading pictures.

Here's a video of how it currently looks like: https://share.getcloudapp.com/qGulx0bm

## Related Tickets & Documents

https://github.com/forem/ForemWebView-ios/issues/17

https://github.com/forem/ForemWebView-ios/pull/18

## QA Instructions, Screenshots, Recordings

1. Open up your local environment and start writing a new Post
2. Image uploads and Cover image uploads should continue to work as expected

It will be tricky to test the native code locally, but you can boot this branch locally and run https://github.com/forem/ForemWebView-ios/pull/18 targeting the local environment. You should then be presented with a native experience to pick images in an Article.

### UI accessibility concerns?

When using the iOS native image upload mode, there is an HTML button to kick start the native image upload experience, so from the web standpoint, launching the native iOS image picker is accessible.

## Added tests?

      have not been included_

## Added to documentation?

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)

## [optional] Are there any post deployment tasks we need to perform?

None

## [optional] What gif best describes this PR or how it makes you feel?

![wicked](https://media.giphy.com/media/6pSJGZW0ZmbUA/giphy.gif)
